### ISSUE TITLE
Minimize dependencies in backend

### ISSUE THREAD

**Issue by Huite** [Deltares/QGIS-Tim#128]
TimML only requires:

"numpy>=1.20", "scipy>=0.19", "numba>=0.39", "matplotlib>=3.0"

TTim only requires:

"numpy>=1.17", "scipy>=1.5", "numba>=0.5", "matplotlib>=3.1", "lmfit>=1.0", "pandas>=1.1"

With the validation refactoring, geopackage reading is no longer required, so geopandas is no longer required.
To easily write data, I suggest using netCDF3 instead of netCDF4, since this only requires scipy which is a requirement anyway. QGIS has netCDF3 support through the inclusion of the netCDF library.

Xarray is still convenient, and only requires:

"numpy>=1.21",  "packaging>=21.3",  "pandas>=1.4"

So only packaging is additionally needed, which is not problematic. All of this is easily pip installed. (Although it still results in a 500 MB installation.)

Things which require some additional research: 

* write the CRS data without rioxarray? (Since rioxarray requires rasterio -> GDAL.)
* See if there's a simple vector format if possible

**Comment by Huite** [comment]
CRS could be omitted: just defer to the project settings as the elements do.

For vector geometries, this package looks extremely helpful:
https://github.com/geomet/geomet

It allows conversion to both geopackage binary, as well as ordinary wkt.

The WKT form is very easy to include: just convert the coordinates into a WKT string and attach it to a pandas dataframe. Then writing the CSV creates a file that QGIS accepts. Geometries cannot be mixed, but that's not very desirable anyway since we'd want to write a separate table per element anyway.

Alternatively, we could create a geopackage which contains all the layers instead. Since geomet converts to the expected binary form, and sqlite3 is part of Python's stdlib and pandas has a straightforward `to_sql` method, this is worth investigating. It is potentially a bit more complicated, especially if something changes in the geopackage specs in the future (arguably not that likely).

**Comment by Huite** [comment]
Fortunately, geopackage appears to be a very sane format!

This suffices to get something readable by QGIS:

```python
import sqlite3

import pandas as pd
from geomet import geopackage

# Some constants
WGS84_WKT = 'GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AXIS["Latitude",NORTH],AXIS["Longitude",EAST],AUTHORITY["EPSG","4326"]]'
APPLICATION_ID = 1196444487
USER_VERSION = 10200


def create_gpkg_contents() -> pd.DataFrame:
    return pd.DataFrame(data={
        "table_name": ["timml Domain:Domain"],
        "data_type": ["features"],
        "identifier": ["timml Domain:Domain"],
        "description": "",
        "last_change": pd.Timestamp.now(),
        "min_x": [20.0],
        "min_y": [10.0],
        "max_x": [170.0],
        "max_y": [140.0],
        "srs_id": [4326],
    })
    

def create_gkpg_spatial_ref_sys() -> pd.DataFrame:
    return pd.DataFrame(data={
        "srs_name": ["Undefined Cartesian SRS", "Undefined geographic SRS", "WGS 84 geodetic"],
        "srs_id": [-1, 0, 4326],
        "organization": ["NONE", "NONE", "EPSG"],
        "organization_coordsys_id": [-1, 0, 4326],
        "definition": ["undefined", "undefined", WGS84_WKT],
        "description": [
            "undefined Cartesian coordinate reference system",
            "undefined geographic coordinate reference system",
            "longitude/latitude coordinates in decimal degrees on the WGS 84 spheroid",
        ]
    })
 

def create_gpkg_geometry_columns() -> pd.DataFrame:
    return pd.DataFrame(data={
        "table_name": ["timml Domain:Domain"],        
        "column_name": "geom",
        "geometry_type_name": ["POLYGON"],
        "srs_id": [4326],
        "z": 0,
        "m": 0,
    })


# Setup data
gpkg_contents = create_gpkg_contents()
gpkg_geometry_columns = create_gpkg_geometry_columns()
gpkg_spatial_ref_sys = create_gkpg_spatial_ref_sys()

polygon = {'type': 'Polygon', 'coordinates': [[[20.0, 20.0], [34.0, 124.0], [70.0, 140.0], [130.0, 130.0], [70.0, 100.0], [110.0, 70.0], [170.0, 20.0], [90.0, 10.0], [20.0, 20.0]]], 'bbox': (20.0, 170.0, 10.0, 140.0)}
binary_geom = geopackage.dumps(polygon)
timml_domain = pd.DataFrame(
    data={"fid": [1], "order": [4], "geom": [binary_geom]}
)

# Write to database.
connection = sqlite3.connect(database="minimum.gpkg")
connection.execute(f"PRAGMA application_id = {APPLICATION_ID};")
connection.execute(f"PRAGMA user_version = {USER_VERSION};")

gpkg_contents.to_sql("gpkg_contents", con=connection)
gpkg_spatial_ref_sys.to_sql("gpkg_spatial_ref_sys", con=connection)
gpkg_geometry_columns.to_sql("gpkg_geometry_columns", con=connection)
timml_domain.to_sql("timml Domain:Domain", con=connection)
```

It doesn't seem really worth to minimize geomet, it's pure Python (and quite clean) and we use a fairly large part anyway.

**Comment by Huite** [comment]
Will be closed by: 3c05195149866cfe130896aa69e93d82a7b00d73
